src/pages-and-resources/discussions/app-list/AppList.jsx

```diff
@@ -161,14 +161,19 @@ const AppList = ({ intl }) => {
       </Responsive>
       <AlertModal
         title={intl.formatMessage(messages.hideDiscussionTabTitle)}
-        isOpen={enabled && !discussionEnabled}
+        isOpen={enabled && !toggleDiscussionEnabledAlert}
         onClose={handleClose}
         isBlocking
         className="hide-discussion-modal"
         footerNode={(
           <ActionRow>
-            <Button variant="link" className="text-decoration-none bg-black" onClick={handleClose}>Cancel</Button>
-            <Button variant="primary" className="bg-primary-500 ml-1 rounded-0" onClick={handleOk}>OK</Button>
+            <Button variant="link" className="text-decoration-none bg-black" onClick={handleClose}>
+              {intl.formatMessage(messages.hideDiscussionCancelButton)}
+            </Button>
+            <Button variant="primary" className="bg-primary-500 ml-1 rounded-0" onClick={handleOk}>
+              {intl.formatMessage(messages.hideDiscussionOkButton)}
+              {saveStatus === SAVING && <Spinner animation="border" size="sm" className="ml-2" />}
+            </Button>
```

Comment:
Use this component
https://paragon-openedx.netlify.app/components/statefulbutton/